src/eth/eip_155_transaction.rs

```diff
@@ -83,14 +75,42 @@ impl EthLegacyEip155TxArgs {
         );
         let v = BigInt::from_bytes_be(Sign::Plus, signature.bytes().get(65..).expect("infallible"));
 
-        validate_eip155_chain_id(eth_chain_id, &v)?;
+        validate_eip155_chain_id(self.chain_id, &v)?;
 
         self.r = r;
         self.s = s;
         self.v = v;
 
         Ok(self)
     }
+
+    pub fn rlp_signed_message(&self) -> anyhow::Result<Vec<u8>> {
```

Comment:
Can we have some tests for this (and other transaction types)? Can be in a different PR. The magic number makes it scary, so I'd like to ensure this doesn't brake if someone adds a new field.

Reply:
This has been covered by `fn test_eth_hash_*`